src/Formats/JSONEachRowUtils.cpp

```diff
@@ -92,6 +100,167 @@ static std::pair<bool, size_t> fileSegmentationEngineJSONEachRowImpl(ReadBuffer
     return {loadAtPosition(in, memory, pos), number_of_rows};
 }
 
+template <const char opening_bracket, const char closing_bracket>
+static String readJSONEachRowLineIntoStringImpl(ReadBuffer & in)
+{
+    skipWhitespaceIfAny(in);
+
+    if (in.eof())
+        throw Exception(ErrorCodes::INCORRECT_DATA, "Cannot read JSON object: unexpected end of file");
+
+    char * pos = in.position();
+    if (*pos != opening_bracket)
+        throw Exception(ErrorCodes::INCORRECT_DATA, "Cannot read JSONEachRow line: {} expected, {} got", opening_bracket, *in.position());
+    ++pos;
+
+    Memory memory;
+    size_t balance = 1;
+    bool quotes = false;
+    while (loadAtPosition(in, memory, pos) && balance)
+    {
+        if (quotes)
+        {
+            pos = find_first_symbols<'\\', '"'>(pos, in.buffer().end());
+
+            if (pos == in.buffer().end())
+                continue;
+
+            if (*pos == '\\')
+            {
+                ++pos;
+                if (loadAtPosition(in, memory, pos))
+                    ++pos;
+            }
+            else if (*pos == '"')
+            {
+                ++pos;
+                quotes = false;
+            }
+        }
+        else
+        {
+            pos = find_first_symbols<opening_bracket, closing_bracket, '\\', '"'>(pos, in.buffer().end());
+
+            if (pos == in.buffer().end())
+                continue;
+
+            else if (*pos == opening_bracket)
+            {
+                ++balance;
+                ++pos;
+            }
+            else if (*pos == closing_bracket)
+            {
+                --balance;
+                ++pos;
+            }
+            else if (*pos == '\\')
+            {
+                ++pos;
+                if (loadAtPosition(in, memory, pos))
+                    ++pos;
+            }
+            else if (*pos == '"')
+            {
+                quotes = true;
+                ++pos;
+            }
+        }
+    }
+
+    if (balance)
+        throw Exception(ErrorCodes::INCORRECT_DATA, "Cannot read JSON object: unexpected end of file");
+
+    saveUpToPosition(in, memory, pos);
+    return String(memory.data(), memory.size());
+}
+
+DataTypePtr getDataTypeFromJSONField(const Poco::Dynamic::Var & field)
```

Comment:
Consider to use `simdjson` from contrib.

Reply:
Will take a look, thanks